src/main/java/pl/pandziak/ticketbookingapp/domain/screening/ScreeningFacadeImpl.java

```diff
@@ -0,0 +1,134 @@
+
+        return toScreeningResponse(screening);
+    }
+
+    @Override
+    public void createScreening(ScreeningRequestDto screeningRequest) {
+        screeningRepository.save(new Screening.Builder()
+                .withId(UUID.randomUUID().toString())
+                .withMovieTitle(screeningRequest.getMovieTitle())
+                .withScreeningTime(screeningRequest.getScreeningTime())
+                .withRoom(Room.valueOf(screeningRequest.getRoom()))
+                .build());
+    }
+
+    @Override
+    public void bookScreeningSeats(String screeningId, TicketsDto tickets) {
+        Screening screening = screeningRepository.findById(screeningId)
+                .orElseThrow(() -> new ScreeningNotFoundException(screeningId));
+
+        screening.bookSeats(tickets.toSeats());
+        screeningRepository.save(screening);
+    }
+
+    @Override
+    public void unbookScreeningSeats(String screeningId, TicketsDto tickets) {
+        Screening screening = screeningRepository.findById(screeningId)
+                .orElseThrow(() -> new ScreeningNotFoundException(screeningId));
+
+        cancelBooking(tickets, screening);
+        screeningRepository.save(screening);
+    }
+
+    private Predicate<? super Screening> startsAtTheSameHourOrLaterThatDay(LocalDateTime date) {
+        LocalDateTime truncated = date.truncatedTo(ChronoUnit.HOURS);
+        return s -> s.getScreeningTime().isEqual(truncated)
+                || s.getScreeningTime().isAfter(truncated);
+    }
+
+    private ScreeningsResponseDto toScreeningsResponse(List<Screening> screenings) {
+        return new ScreeningsResponseDto(
+                screenings.stream()
+                        .map(screening -> new ScreeningDetailsDto.Builder()
+                                .withScreeningId(screening.getId())
+                                .withMovieTitle(screening.getMovie())
+                                .withScreeningTime(screening.getScreeningTime())
+                                .withRoom(screening.getRoom())
+                                .build())
+                        .collect(toList())
+        );
+    }
+
+    private ScreeningResponseDto toScreeningResponse(Screening screening) {
+        return new ScreeningResponseDto(
+                new ScreeningDetailsDto.Builder()
+                        .withScreeningId(screening.getId())
+                        .withMovieTitle(screening.getMovie())
+                        .withScreeningTime(screening.getScreeningTime())
+                        .withRoom(screening.getRoom())
+                        .build(),
+                prepareSeatsLayoutForRoom(screening)
+        );
+    }
+
+    private List<SeatDto> prepareSeatsLayoutForRoom(Screening screening) {
+        List<Seat> seatsTaken = screening.getSeatsTaken();
+        Room room = screening.getRoom();
+
+        List<SeatDto> result = new LinkedList<>();
+
+        for (int i = 1; i <= room.getRows(); i++) {
```

Comment:
a jak to napisać w stylu nowszych niż 7 wersji javy?

Reply:
Wydaje mi się, że to rozwiązanie jest prostsze i czytelniejsze. Gdybym miała użyć Javy 8 i streamów, pewnie użyłabym dwóch IntStreamów i odpowiedniego mapowania na wartości, ale mam wrażenie że wtedy kod staje się mniej czytelny.